src/crimson/osd/osd_operations/snaptrim_event.cc

```diff
@@ -148,6 +149,11 @@ SnapTrimEvent::with_pg(
             pg,
             object,
             snapid);
+          if (pg->get_peering_state().state_test(PG_STATE_SNAPTRIM_ERROR)) {
```

Comment:
As far as I recall the original idea was to minimize the # communication channels and keep the flag management in the client of `SnapTrimEvent`. It already does:

```cpp
        return shard_services.start_operation<SnapTrimEvent>(
          this,
          snap_mapper,
          to_trim,
          needs_pause
        ).second.handle_error(
          crimson::ct_error::enoent::handle([this] {
            logger().error("{}: ENOENT saw, trimming stopped", *this);
            peering_state.state_set(PG_STATE_SNAPTRIM_ERROR);
            publish_stats_to_osd();
            return seastar::make_ready_future<seastar::stop_iteration>(
              seastar::stop_iteration::yes);
          }), 
```

Here `PG_STATE_SNAPTRIM_ERROR` is being used as a side-channel to the error-in-return-future.
Why not let `SnapTrimObjSubEvent` simply returned a failed `remove_or_update_iertr::future<>`?

Reply:
The error is now passed_further to be handled by the  the client of `SnapTrimEvent`.